solr/core/src/java/org/apache/solr/update/processor/MinFieldValueUpdateProcessorFactory.java

```diff
@@ -55,20 +55,19 @@
 public final class MinFieldValueUpdateProcessorFactory extends FieldValueSubsetUpdateProcessorFactory {
 
   @Override
-  @SuppressWarnings({"unchecked"})
-  public Collection<Object> pickSubset(@SuppressWarnings({"rawtypes"})Collection values) {
-    @SuppressWarnings({"rawtypes"})
-    Collection result = values;
+  public <T> Collection<T> pickSubset(Collection<T> values) {
     try {
+        // TODO: Verify this
       // NOTE: the extra cast to Object is needed to prevent compile
       // errors on Eclipse Compiler (ecj) used for javadoc lint
```

Comment:
minor: once the verification is done the comment w.r.t. the extra cast would become removable
```suggestion
        // TODO: Verify this
```